actions/weather_MOD.ts

```diff
@@ -0,0 +1,215 @@
+
+  static html(_isEvent: any, data: any) {
+    return `
+<div style="float: left; width: 55%; padding-top: 8px;">
+  Source City:<br>
+  <input id="city" class="round" type="text">
+  </div><br>
+<div style="float: left; width: 100%; padding-top: 8px;">
+  Source Info:<br>
+  <select id="info" class="round">
+    <option value="0" selected>City Name</option>
+    <option value="1">Region</option>
+    <option value="2">Country</option>
+    <option value="3">Latitude</option>
+    <option value="4">Longitude</option>
+    <option value="5">Timezone</option>
+    <option value="6">Date</option>
+    <option value="7">Temperature (Celsius)</option>
+    <option value="8">Temperature (Fahrenheit)</option>
+    <option value="9">Feels Like (Celsius)</option>
+    <option value="10">Feels Like (Fahrenheit)</option>
+    <option value="11">Condition</option>
+    <option value="12">Condition Icon URL</option>
+    <option value="13">Wind Speed (Kilometer per hour)</option>
+    <option value="14">Wind Speed (Mile per hour)</option>
+    <option value="15">Wind Direction</option>
+    <option value="16">Pressure (hPa)</option>
+    <option value="17">Pressure (psi)</option>
+    <option value="18">Humidity (Percent)</option>
+    <option value="19">Cloud (Percent)</option>
+    <option value="20">Visibility (Kilometer)</option>
+    <option value="21">Visibility (Mile)</option>
+    <option value="22">UV Index</option>
+    <option value="23">Gust (Kilometer per hour)</option>
+    <option value="24">Gust (Mile per hour)</option>
+  </select>
+</div><br>
+<div>
+  <div style="float: left; width: 35%; padding-top: 8px;">
+    Store In:<br>
+    <select id="storage" class="round" onchange="glob.variableChange(this, 'varNameContainer')">
+      ${data.variables[0]}
+    </select>
+  </div>
+  <div id="varNameContainer" style="float: right; width: 60%; padding-top: 8px;">
+    Variable Name:<br>
+    <input id="varName" class="round" type="text"><br>
+  </div>
+</div>`;
+  }
+
+  static init(this: Actions) {
+    const { glob, document } = this;
+    glob.variableChange(document.getElementById('storage'), 'varNameContainer');
+  }
+
+  static action(this: Actions, cache: ActionCache) {
+    const data = cache.actions[cache.index];
+    const info = parseInt(data.info, 10);
+    const city = this.evalMessage(data.city, cache);
+
+    if (!city) {
+      console.error('Weather Action: Please specify a city to fetch the weather information.');
+      return this.callNextAction(cache);
+    }
+
+    const Mods = this.getMods();
+    const fetch = Mods.require('node-fetch');
+    fetch(`https://fnrr.dev/weather/${encodeURI(city)}`)
+      .then((result: any) => result.json())
+      .then((data: any) => {
+        if (data?.error !== undefined) {
+          console.error(
+            `Weather Action: There was an error while fetching the weather info: ${
+              typeof data.error === 'string' ? data.error : data?.message
+            }`,
+          );
```

Comment:
data.error can return either string (explains the reason it failed) or boolean value. When it is boolean, message property (explains the reason it failed) is there